### PR TITLE
Abstract API for AI model clients and portable client request options design.

### PR DESCRIPTION
**Pull request by markpollack**
This is a squahed commit of the 'options-ftw' comment.

Short story is this PR provides

* An abstract API for AI model clients
* Providing portable client request options while still allowing vendor specific options when required.  Implemented only for StabilityAI/OpenAI ImageClient

Open TODOs are to

* implement the options design pattern across the code base

Look at the commit a0f71f7c9bc87dba73234cac41c1cc040946af66 for the details of the squashed commits that went into this PR

### CODE REVIEW

File: spring-ai-core/src/main/java/org/springframework/ai/image/NewImageClient.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.image;
+
+import org.springframework.ai.model.ModelClient;
+
+@FunctionalInterface
+public interface NewImageClient extends ModelClient<ImagePrompt, ImageResponse> {
```
Comment (tzolov) [inline spring-ai-core/src/main/java/org/springframework/ai/image/NewImageClient.java:22]: This seems redundant as it is exactly like ImageClient?

File: models/spring-ai-openai/src/test/java/org/springframework/ai/openai/image/OpenAiImageClientIT.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.springframework.ai.image.ImagePrompt;
+import org.springframework.ai.image.ImageResponse;
+import org.springframework.ai.openai.OpenAiTestConfiguration;
+import org.springframework.ai.openai.testutils.AbstractIT;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest(classes = OpenAiTestConfiguration.class)
+@EnabledIfEnvironmentVariable(named = "OPENAI_API_KEY", matches = ".+")
+public class OpenAiImageClientIT extends AbstractIT {
```
[the remark this answers is not in the document]
Comment (markpollack) [reply models/spring-ai-openai/src/test/java/org/springframework/ai/openai/image/OpenAiImageClientIT.java:28]: It exposed a bug, need to figure out what to do on retry when it is an API (400) error that is fatal.  Also, not seeing any logging on retry will make these cases hard to diagnose in the wild.

File: models/spring-ai-openai/src/test/java/org/springframework/ai/openai/image/OpenAiImageClientIT.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Test
+	void imageAsUrlTest() {
+		ImagePrompt imagePrompt = new ImagePrompt("Create an image of a mini golden doodle dog.");
+
+		ImageResponse imageResponse = openaiImageClient.call(imagePrompt);
+
+		System.out.println(imageResponse);
```
[the remark this answers is not in the document]
Comment (markpollack) [reply models/spring-ai-openai/src/test/java/org/springframework/ai/openai/image/OpenAiImageClientIT.java:36]: Added asserts

**Comment by markpollack** [comment]
merged in 243cef976c6ff6e2f8ae4aaee75cccdde45051a6

**Pull request closed without merging**